experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -328,20 +329,32 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 		kind: RevertibleKind,
 	): Revertible {
 		this.revertibleCommits.set(commit.revision, commit);
+		let discarded = false;
 		const revertible = {
 			kind,
 			origin: {
 				// This is currently always the case, but we may want to support reverting remote ops
 				isLocal: true,
 			},
 			revert: () => {
-				this.revert(commit.revision, kind);
-				return RevertResult.Success;
+				if (discarded) {
+					fail("revertible has already been discarded");
+				}
+				const revertCommit = this.revert(commit.revision, kind);
+				if (revertCommit !== undefined) {
+					revertible.discard();
+					return RevertResult.Success;
+				}
+				return RevertResult.Failure;
 			},
 			discard: () => {
+				if (discarded) {
+					fail("revertible has already been discarded");
+				}
 				// TODO: delete the repair data from the forest
```

Comment:
If this is being called by revert(), the repair data should already be gone?

Reply:
This is for revertibles that don't get reverted